[Fix] create language inherited child didn't set the controller & template

youwe-petervanderwal opened the pull request:
## Changes in this pull request  
Resolves bug #11507

## Additional info  
Related snippet from DocumentController::addAction():

```
$docType = Document\DocType::getById($request->get('docTypeId'));
if ($docType) {
   // ...
} elseif ($request->get('translationsBaseDocument')) {
    // This section was never reached because $docType was always a DocType object
    $translationsBaseDocument = Document::getById($request->get('translationsBaseDocument'));
    if ($translationsBaseDocument instanceof Document\PageSnippet) {
        $createValues['template'] = $translationsBaseDocument->getTemplate();
        $createValues['controller'] = $translationsBaseDocument->getController();
    }
}
```

## Note
Seems that this bug is introduced in f008e2b973f1ffbad20f982f70230287df1a18d9 - not 100% sure this is the right fix to this but I assume that change was also done for a reason

youwe-petervanderwal committed:
Fix DocType::getById() should return null on empty id

Error: When creating a new document translation variant the controller+template weren't set.

Caused by: DocType::getById() returning a DocType object (with empty properties though) on null value id's. The if-elsif flow in the DocumentController::addAction() then (unjustly) skipped the 'translationsBaseDocument' logic.

Resolved by: Adding an if(empty($id)) check to the getById() method

dvesh3 commented:
@youwe-petervanderwal This PR looks good to me. Could you please also revert this change https://github.com/pimcore/pimcore/pull/10478/files#diff-f7053f28596f2f8ceaf52906b900bceb1c8093e4cc701dafc9b17156a811cadcR51-R55? as Dao::getById() isn't suppose to return value. thanks!

youwe-petervanderwal committed:
Fix DocType\Dao::getById() shouldn't return anything

youwe-petervanderwal commented:
@dvesh3 
- Reverted commit
- Rewritten the logic of that method so that your original goal is still achieved (otherwise it somehow complained on passing null to getDataByName() which wasn't allowed)
- Added argument and void return type to that method

dvesh3 commented:
@youwe-petervanderwal thank you!
